bpf/lib/nodeport.h

```diff
@@ -1276,7 +1278,7 @@ static __always_inline int nodeport_nat_ipv4_fwd(struct __ctx_buff *ctx)
 	};
 	int ret = CTX_ACT_OK;
 
-	if (snat_v4_needed(ctx, &target.addr, &from_endpoint))
+	if (snat_v4_needed(ctx, &target.addr, &from_endpoint, &target.egress_gateway))
```

Comment:
Can we pass the whole `target` as parameter instead, and let `snat_v4_needed()` adjust `.addr` and `.egress_gateway` as needed?